server.js

```diff
@@ -1,39 +1,187 @@
+import listEndpoints from 'express-list-endpoints'
+import dotenv from 'dotenv'
 
-// If you're using one of our datasets, uncomment the appropriate import below
-// to get started!
-// 
-// import goldenGlobesData from './data/golden-globes.json'
-// import avocadoSalesData from './data/avocado-sales.json'
-// import booksData from './data/books.json'
-// import netflixData from './data/netflix-titles.json'
-// import topMusicData from './data/top-music.json'
+import netflixTitles from './data/netflix-titles.json'
+
+dotenv.config()
 
 const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/project-mongo"
 mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
 mongoose.Promise = Promise
 
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
+const titlesSchema = new mongoose.Schema({
+  show_id: String,
+  type: String, 
+  title: String,
+  director: { 
+    type: mongoose.Schema.Types.ObjectId, 
+    ref: 'Director',
+  },
+  cast: String,
+  country: String,
+  date_added: String,
+  release_year: Number,
+  rating: String, 
+  duration: String,
+  listed_in: String,
+  description: String
+})
+
+const directorsSchema = new mongoose.Schema({
+  name: String
+})
+
+const Title = mongoose.model('Title', titlesSchema)
+
+const Director = mongoose.model('Director', directorsSchema)
+
+if (process.env.RESET_DB) {
+  const seedDB = async () => {
+    await Title.deleteMany()
+    await Director.deleteMany()
+
+    let directors = []
+
+    // 800 objects for deployment purpose
+    netflixTitles.slice(0,800).forEach(async item => {  
+      const director = new Director({"name": item.director}) 
+      
+      if (item.director != "") {
+        directors.push(director)
+        await director.save()  
+      } 
+    })
+
+    // 800 objects for deployment purpose
+    netflixTitles.slice(0,800).forEach(async item => { 
+      const title = new Title({
+        ...item,
+        director: directors.find(singleDirector => singleDirector.name === item.director)
+      })
+      await title.save()
+    }) 
+  }
+  seedDB()
+}
+
 const port = process.env.PORT || 8080
 const app = express()
 
-// Add middlewares to enable cors and json body parsing
 app.use(cors())
-app.use(bodyParser.json())
+app.use(express.json())
+
+// check if mongoose connection is alright
+app.use((req, res, next) => {
+  if (mongoose.connection.readyState === 1) {
+    next()
+  } else {
+    res.status(503).json({ error: 'Service unavailable' })
+  }
+})
 
-// Start defining your routes here
 app.get('/', (req, res) => {
-  res.send('Hello world')
+  res.send(listEndpoints(app))
+}) 
+
+// return all titles
+app.get('/titles', async (req, res) => {
+    const titles = await Title.find().populate('director')
+    res.json({ length: titles.length, data: titles })
+})
+
+// query by year
+app.get('/titles/year', async (req, res) => {
```

Comment:
when you use queries to filter your data, then do not create a separate endpoint. use generic GET endpoint for this collection: 

`/titles?year=2000` not `/titles/year?year=2000`